source/use-cases/client-side-field-level-encryption-guide.txt

```diff
@@ -250,15 +262,37 @@ information on the features of envelope encryption and key management
 concepts, see :aws-docs:`AWS Key Management Service Concepts
 <kms/latest/developerguide/concepts.html#enveloping>`.
 
-The master key, used by the MongoDB driver to create and encrypt data
-keys, should be stored remotely in a :wikipedia:`Key Management System
-<Key_management#Key_management_system>`. The data encryption keys,
-generated and used by the MongoDB driver to encrypt and decrypt document
-fields, are stored in a key vault collection in the same database as the
-encrypted data.
+The following diagram shows how the **master key** and **data encryption
+keys** are created and stored in the MongoDB implementation of CSFLE:
+
+.. image:: /figures/CSFLE_Create_Keys_Local.png
+   :alt: Diagram that describes creating master and data encryption keys when using a local provider
+
+In this step, we create and store the master key, used by the MongoDB driver
+to encrypt data encryption keys, in the **Local Key Provider** which is the
+filesystem in our local development environment. We refer to this key as
+the "locally-managed master key" in this guide.
+
+The data encryption keys, generated and used by the MongoDB driver to encrypt
+and decrypt document fields, are stored in a key vault collection in the same
+MongoDB replica set as the encrypted data.
+
+.. admonition:: Local Key Provider is not suitable for production
+   :class: important
+
+   The Local Key Provider is an insecure method of storage and is therefore
+   **not recommended** if you plan to use CSFLE in production. Instead,
+   the master key should be managed in a :wikipedia:`Key Management
```

Comment:
nits:
- active voice: "you should manage"
- we say "manage(d)" a lot here. Maybe we could re-org a bit to cut down? e.g.

```rst
Instead, use a KMS that stores the master key and remotely manages
storage and decryption for data keys.
```

Reply:
Thanks, good catch, will update.